community_supplied/Chassis/netconf-system-cpu-temperatures.rule

```diff
@@ -0,0 +1,147 @@
+                }
+            }
+            sensor chassis-sensor {
+			    synopsis "Chassis temperature sensor initialization";
+                description "iAgent sensor to collect data from network device"; 
+                iAgent {
+                    file chassis-env-info.yml;
+                    table ChassisStatsTable;
+                    frequency 300s;
+                }
+            }
+			/*
+            * Fields defined with variables, dynamic thresholds and constant values
+            */
+            field component-name {
+                sensor chassis-sensor {
+                    path name;
+                }
+                type string;
+                description "Name of the component";
+            }
+            field dt-temperature-value {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$temperature-value";
+                    }
+                }                                   
+                type integer;
+                description "dynamic threshold value for component temperature";
+            }
+            field temperature-value {
+                formula {
+                    user-defined-function {
+                        function-name temp_degrees;
+                        argument temperature "$temperature";
+                    }
+                }
+                type integer;
+            }
+            field threshold {
+                constant {
+                    value "{{threshold-value}}";
+                }
+                type integer;
+                description "Threshold value for component temperature";
+            }
+			/*
+            * Anomaly detection logic for chassis components.
+            */
+            trigger component-temperature {
+			    /*
+                * Sets color to red and sends out an anomaly notification when
+                * temperature value increases more than threshold static ($threshold) value
+                * 
+                */ 
+                frequency 300s;
+                term is-component-temperature-is-abnormal {
+                    when {
+                        greater-than "$temperature-value" "$threshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Temperature of $component-name has crossed static threshold value ($threshold), Value=$temperature-value (degrees)";
+                        }
+                    }
+                }
+				/*
+                * Sets color to yellow and sends out an anomaly notification when
+                * temperature value increases more than dynamic threshold value
+                * 
+                */
+                term is-component-temperature-is-minimal {

```

Comment:
How about has-component-temperature-breached-dynamic-threshold?

Reply:
Modified to "is-component-temperature-above-dt"